### DISCUSSION TITLE
Request for a new Query parameter: Preset/Bookmark id

### DISCUSSION THREAD

**Discussion by arladmin** [directus/directus#19994] [Feature Requests]
### Summary

When calling the List API endpoint, i would like to be able to provide a Preset/Bookmark ID also (whenever possible).

This would allow to save quite some effort when retrieving items which have already been filtered in a saved Preset/Bookmark.

This could also be compared to the 'retrieve records for a particular View' feature in Airtable.

<hr>

### Basic Example

Say, a Preset exists, for the invoking user, called 'Restricted' with ID = 2.
The Preset has certain filters, e.g. 
- status _is not_ 'draft' 
- is_flagged _equals_ true

Now, be able to make the following API call:

1.  `GET /items/articles?preset=2`
2.  `GET /items/articles?preset=Restricted`

<hr>

### Motivation

Currently, a lot of time is spent in creating the filters, especially when there are too many conditions, when it could be avoided by simply providing the id of a saved Preset/Bookmark, so that the saved filter configuration gets passed on to the API call automatically.

<hr>

### Detailed Design

1. Add a new Query Parameter, called e.g. Preset
3. The user would then have to simply provide the ID of the Preset
4. The Filter Parameter should still be able to pass on additional filters, just like in Directus Studio

<hr>

### Must Have:
- The ability to provide the Preset ID

### Could Have:
- The ability to provide the name of the Preset instead of the ID

<hr>

### Drawbacks

Can't think of any drawbacks.

<hr>

### Alternatives

The current workaround would be to first retrieve a Preset by its value, and then pass on the obtained filter value to the concerned Items retrieval API call.

<hr>

### Adoption Strategy

I imagine we could simply add another Query Parameter, called Preset.

<hr>

**Comment by arladmin** [comment]
The above feature could also help, to certain extent, with this FR: https://github.com/directus/directus/discussions/19753#discussion-5656653

**Comment by ghost** [comment]
Let’s say you retrieve data from the API where you use this `preset` parameter and you use it in a real world app or website: What if the preset gets deleted?
This could be dangerous!

**Comment by arladmin** [comment]
@JonathanDoelan 
> Let’s say you retrieve data from the API where you use this `preset` parameter and you use it in a real world app or website: What if the preset gets deleted? This could be dangerous!

Hi. Sorry, but couldn't understand why that's a special case?
Couldn't there be a lot of workaround for that? e.g.
- Add a 'tag' to the preset
- Prompt before deleting
- Be able to provide the 'unique' name of the preset instead of the ID, so that, even if a new one's created with the same name, it works

**Reply by ghost** [reply]
Thanks for your reply. That’s an important information. Prompt before deletion and make clear that it could be used in an API request would make sense!

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
